specs/experimental/op-contracts-manager.md

```diff
@@ -20,146 +20,271 @@ of governance approved [contract releases] can be found on the
-in the [standard configuration]. This improves UX by removing an input, and ensures uniqueness of
-the batch inbox addresses.
+Any contracts which do not meet this requirement will need to be deployed by the `upgrade()`
+function, increasing the cost and reducing the number of OP Chains which can be atomically upgraded.
 
-### Contract Deployments
+## Adding game types
 
-All contracts deployed by the OP Contracts Manager are deployed with CREATE2, with a
-salt equal to either:
+Because different OP Chains within a Superchain may use different dispute game types, and are
+expected to move from a permissioned to permissionless game over time, an `addGameType()` method is
+provided to enable adding a new game type to multiple games at once.
 
-- The L2 chain ID, or
-- `keccak256(bytes.concat(bytes32(uint256(l2ChainId)), contractName))`.
+### Interface
 
-The former is used when only a single instance of a given contract is deployed for a chain.
-The latter is used when deploying multiple instances of a given contract for a chain,
-which is the case of all `Proxy` contracts. For these, the `contractName`
-is the name of the implementation contract that will be used with the proxy.
+#### `addGameType`
 
-This provides the following benefits:
+The `addGameType` method is used to orchestrate the actions required to add a new game type to one
+or more chains.
 
-- Contract addresses for a chain can be derived as a function of chain ID without any RPC calls.
-- Chain ID uniqueness is enforced for free, as a deploy using the same chain ID
-will result in attempting to deploy to the same address, which is prohibited by
-the EVM.
-  - This property is contingent on the proxy and `AddressManager` code not
-  changing when OP Contracts Manager is upgraded. Both of these are not planned to
-  change.
-  - The OP Contracts Manager is not responsible for enforcing chain ID uniqueness, so it is acceptable
-  if this property is not preserved in future versions of the OP Contracts Manager.
+```solidity
+struct PermissionlessGameConfig {
+  bytes32 absolutePrestate;
+}
+
+function addGameType(ISystemConfig[] _systemConfigs, PermissionlessGameConfig[] _newGames) public;
+```
+
+### Implementation
+
+The high level logic of the `addGameType` method is as follows (for each chain):
+
+1. Deploy and initialize new `DelayedWethProxy` for the new game type, reusing the existing implementation
+1. Deploy a new `FaultDisputeGame` contract. The source of the constructor args is indicated below this list,
+   the value which is not available onchain is the `absolutePrestate`.
+1. Calls `upgrade()` on the `AnchorStateRegistry` to set the new game type to
+   add a new entry to the `anchors` mapping. The `upgrade()` method should
+   revert if it would overwrite an existing entry.
+1. Read the `DisputeGameFactory` address from the `SystemConfig`.
+1. Call `DisputeGameFactory.setImplementation()` to register the new game.
+
+```solidity
```

Comment:
This seems better for a table with name, type and description columns

Reply:
good call

Reply:
19cf6cf2a2c5